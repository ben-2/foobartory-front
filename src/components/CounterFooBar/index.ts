import { CounterFooBar } from "./CounterFooBar";
export default CounterFooBar;