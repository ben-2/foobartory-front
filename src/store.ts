import { createStore, Action, action } from 'easy-peasy';
import { IRobotAttributes } from './store.type';
export interface IStoreModel {
  lang: string;
  setLang: Action<IStoreModel, string>;
  isGameStarted: boolean;
  startGame: Action<IStoreModel>;
  robotsConfiguration: IRobotAttributes[];
  countFoo: number;
  countBar: number;
  countFooBar: number;
}

const model: IStoreModel = {
  lang: 'fr',
  setLang: action((state, payload) => {
    state.lang = payload;
  }),
  isGameStarted: false,
  startGame: action((state, payload) => {
    state.isGameStarted = true;
  }),
  robotsConfiguration: [
    {
      id: 1,
      currentActivity: '',
      isRobotAvailable: true,
      previousActivity: '',
    },
    {
      id: 2,
      currentActivity: '',
      isRobotAvailable: true,
      previousActivity: '',
    }
  ],
  countFoo: 0,
  countBar: 0,
  countFooBar: 0
};


export const store = createStore<IStoreModel>(model);
