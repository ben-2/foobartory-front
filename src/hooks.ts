import { createTypedHooks } from 'easy-peasy';
import { IStoreModel } from './store';

const { useStoreActions, useStoreState, useStoreDispatch, useStore } = createTypedHooks<IStoreModel>();

export {
  useStoreActions,
  useStoreState,
  useStoreDispatch,
  useStore
};