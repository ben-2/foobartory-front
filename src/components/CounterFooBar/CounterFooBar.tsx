import { useStoreState } from '../../hooks';

export function CounterFooBar() {
  const countFooBar = useStoreState((state) => state.countFooBar);

  return (
    <>
      {countFooBar}
    </>
  );
}
