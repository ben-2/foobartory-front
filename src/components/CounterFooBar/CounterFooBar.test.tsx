import { createStore, StoreProvider } from 'easy-peasy';
import { CounterFooBar } from './CounterFooBar';
import { store } from '../../store';
import { mount } from 'enzyme';
import { useStoreState } from '../../hooks';

describe('<CounterFooBar />', () => {
  it('match snapshot', () => {
    const initialStateForTest = {
      countFooBar: 0
    };
    const storeTest = createStore(store, { initialState: initialStateForTest });
    const wrapper = mount(
      <StoreProvider store={storeTest}>
        <CounterFooBar />
      </StoreProvider>)
    expect(wrapper).toMatchSnapshot();
  });

  it('counter to display 11', () => {
    const initialStateForTest = {
      countFooBar: 11
    };
    const storeTest = createStore(store, { initialState: initialStateForTest });
    const wrapper = mount(
      <StoreProvider store={storeTest}>
        <CounterFooBar />
      </StoreProvider>)
    expect(wrapper.text()).toEqual('11');
  });

  it('counter to display 55', () => {
    const initialStateForTest = {
      countFooBar: 55
    };
    const storeTest = createStore(store, { initialState: initialStateForTest });
    const wrapper = mount(
      <StoreProvider store={storeTest}>
        <CounterFooBar />
      </StoreProvider>)
    expect(wrapper.text()).toEqual('55');
  });
});
