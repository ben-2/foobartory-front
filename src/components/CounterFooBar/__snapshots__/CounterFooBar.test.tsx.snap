// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterFooBar /> match snapshot 1`] = `ReactWrapper {}`;
